index.js

```diff
@@ -5,6 +5,10 @@ var Parser = acorn.Parser;
 var pp = Parser.prototype;
 
 function CustomParser(input) {
+	// state
+	this.imports = [];
+	this.inputSource = input;
+
```

Comment:
This is rather unnecessary here - acorn already provides `this.input` for the source, and as for `imports` - if we don't do live bindings anyway, can generate them in-place as variable destructurings (I'm sure that in 99% of cases `import`s are at the top of the file anyway).